File: sites/arif-fazil.com/patches/vite-plugin-ssg.py
#!/usr/bin/env python3
"""Patch vite-plugin-ssg dist/plugin.js for Tailwind v3 compatibility."""
import os, sys

plugin_path = sys.argv[1] if len(sys.argv) > 1 else "node_modules/vite-plugin-ssg/dist/plugin.js"
if not os.path.exists(plugin_path):
    print(f"[patch] {plugin_path} not found, skipping")
    sys.exit(0)

with open(plugin_path) as f:
    content = f.read()

if "dummy-tailwind" in content:
    print("[patch] vite-plugin-ssg already patched, skipping")
    sys.exit(0)

patches = 0

# Patch 1: SSR renderer - replace @tailwindcss/vite with dummy, add alias
old = 'var tailwindcss = require2("@tailwindcss/vite").default;\nasync function renderPage(options) {\n  const { componentPath, routeUrl, islands, renderHead: shouldRenderHead, config } = options;\n  const serverEntryPath = path3.join(__dirname, "..", "src", "server-entry.tsx");\nconst vite = await createServer({\n    configFile: false,\n    // Don\'t load vite.config.ts\n    root: process.cwd(),\n    // Use project\'s root to resolve node_modules\n    plugins: [reactPlugin(), tailwindcss(), ...config.vite.plugins],'
new = 'var tailwindcss = () => ({ name: "dummy-tailwind", enforce: "pre" });\nasync function renderPage(options) {\n  const { componentPath, routeUrl, islands, renderHead: shouldRenderHead, config } = options;\n  const serverEntryPath = path3.join(__dirname, "..", "src", "server-entry.tsx");\nconst _alias = { "@": path3.resolve(process.cwd(), "src") };\nconst vite = await createServer({\n    configFile: false,\n    // Don\'t load vite.config.ts\n    root: process.cwd(),\n    // Use project\'s root to resolve node_modules\n    // Rely on postcss.config.js for Tailwind CSS processing\n    plugins: [reactPlugin(), ...config.vite.plugins],\n    resolve: { alias: _alias },'
if old in content:
    content = content.replace(old, new)
    patches += 1
    print("[patch] SSR renderer patched")
else:
    print("[patch] WARNING: SSR renderer pattern not found")

# Patch 2: CSS minifier lightningcss -> esbuild
if "cssMinify: config.css.minify," in content:
    content = content.replace("cssMinify: config.css.minify,", 'cssMinify: "esbuild",')
    patches += 1
    print("[patch] CSS minifier patched")
else:
    print("[patch] WARNING: CSS minifier pattern not found")

# Patch 3: CSS bundle builder - remove tailwindcss2, add alias
old3 = 'var tailwindcss2 = require3("@tailwindcss/vite").default;'
new3 = 'var tailwindcss2 = () => ({ name: "dummy-tailwind", enforce: "pre" });'
if old3 in content:
    content = content.replace(old3, new3)
    patches += 1
    print("[patch] CSS builder tailwindcss2 patched")
else:
    print("[patch] WARNING: CSS builder tailwindcss2 pattern not found")

old3b = 'const plugins = [\n      reactPlugin2(),\n      tailwindcss2(),\n      ...config.vite.plugins\n    ];'
new3b = 'const _cssAlias = { "@": path5.resolve(process.cwd(), "src") };\n    const plugins = [\n      reactPlugin2(),\n      ...config.vite.plugins\n    ];'
if old3b in content:
    content = content.replace(old3b, new3b)
    patches += 1
    print("[patch] CSS builder plugins patched")
else:
    print("[patch] WARNING: CSS builder plugins pattern not found")

# Add resolve alias to CSS builder vite config
old3c = 'plugins,\n      build: {'
new3c = 'plugins,\n      resolve: { alias: _cssAlias },\n      build: {'
if old3c in content:
    content = content.replace(old3c, new3c)
    patches += 1
    print("[patch] CSS builder resolve alias patched")

# Patch 4: Hydration builder - remove tailwindcss3
old4 = 'var tailwindcss3 = require4("@tailwindcss/vite").default;'
new4 = 'var tailwindcss3 = () => ({ name: "dummy-tailwind", enforce: "pre" });'
if old4 in content:
    content = content.replace(old4, new4)
    patches += 1
    print("[patch] Hydration builder tailwindcss3 patched")
else:
    print("[patch] WARNING: Hydration tailwindcss3 pattern not found")

old4b = 'plugins: [\n      reactPlugin3(),\n      tailwindcss3(),\n      ...config.vite.plugins\n    ];'
new4b = 'plugins: [\n      reactPlugin3(),\n      ...config.vite.plugins\n    ];'
if old4b in content:
    content = content.replace(old4b, new4b)
    patches += 1
    print("[patch] Hydration builder plugins patched")
else:
    print("[patch] WARNING: Hydration plugins pattern not found")

with open(plugin_path, 'w') as f:
    f.write(content)

print(f"[patch] Applied {patches} patches to vite-plugin-ssg")

File: sites/arif-fazil.com/patches/vite-plugin-ssg.js
#!/usr/bin/env node
/**
 * Patches vite-plugin-ssg dist/plugin.js for Tailwind v3 compatibility.
 * Run via postinstall after npm install.
 */
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const pluginPath = path.join(__dirname, '..', 'node_modules', 'vite-plugin-ssg', 'dist', 'plugin.js');

if (!fs.existsSync(pluginPath)) {
  console.log('[patch] vite-plugin-ssg not found, skipping');
  process.exit(0);
}

let content = fs.readFileSync(pluginPath, 'utf8');

if (content.includes('dummy-tailwind')) {
  console.log('[patch] vite-plugin-ssg already patched, skipping');
  process.exit(0);
}

let patches = 0;

// Patch 1: SSR renderer - replace @tailwindcss/vite with dummy
const r1 = content.match(/var tailwindcss = require2\("@tailwindcss\/vite"\)\.default;/);
if (r1) {
  content = content.replace(r1[0], 'var tailwindcss = () => ({ name: "dummy-tailwind", enforce: "pre" });');
  patches++;
  console.log('[patch] SSR tailwindcss patched');
}

// Add @ alias after createServer root
content = content.replace(
  /const vite = await createServer\(\{\s*configFile: false,\s*root: process\.cwd\(\),/,
  `const _alias = { "@": path3.resolve(process.cwd(), "src") };\n    const vite = await createServer({\n    configFile: false,\n    root: process.cwd(),`
);

// Add resolve.alias to SSR vite config
content = content.replace(
  /plugins: \[reactPlugin\(\), \.\.\.config\.vite\.plugins\],/,
  'plugins: [reactPlugin(), ...config.vite.plugins],\n    resolve: { alias: _alias },'
);

// Patch 2: CSS builder - replace @tailwindcss/vite with dummy
const r2 = content.match(/var tailwindcss2 = require3\("@tailwindcss\/vite"\)\.default;/);
if (r2) {
  content = content.replace(r2[0], 'var tailwindcss2 = () => ({ name: "dummy-tailwind", enforce: "pre" });');
  patches++;
  console.log('[patch] CSS builder tailwindcss2 patched');
}

// Remove tailwindcss2() from CSS builder plugins
content = content.replace(
  /plugins: \[\n?      reactPlugin2\(\),\n?      tailwindcss2\(\),\n?      \.\.\.config\.vite\.plugins\n?    \];/,
  'plugins: [\n      reactPlugin2(),\n      ...config.vite.plugins\n    ];'
);

// Add resolve alias to CSS builder vite config
content = content.replace(
  /rollupOptions: \{/,
  `resolve: { alias: { "@": path5.resolve(process.cwd(), "src") } },\n      rollupOptions: {`
);

// Patch 3: CSS minifier lightningcss -> esbuild
if (content.includes('cssMinify: config.css.minify,')) {
  content = content.replace('cssMinify: config.css.minify,', 'cssMinify: "esbuild",');
  patches++;
  console.log('[patch] CSS minifier patched');
}

// Patch 4: Hydration builder - replace @tailwindcss/vite with dummy
const r3 = content.match(/var tailwindcss3 = require4\("@tailwindcss\/vite"\)\.default;/);
if (r3) {
  content = content.replace(r3[0], 'var tailwindcss3 = () => ({ name: "dummy-tailwind", enforce: "pre" });');
  patches++;
  console.log('[patch] Hydration tailwindcss3 patched');
}

// Remove tailwindcss3() from hydration plugins
content = content.replace(
  /plugins: \[\n?      reactPlugin3\(\),\n?      tailwindcss3\(\),\n?      \.\.\.config\.vite\.plugins\n?    \];/,
  'plugins: [\n      reactPlugin3(),\n      ...config.vite.plugins\n    ];'
);

fs.writeFileSync(pluginPath, content);
console.log(`[patch] Applied ${patches} patches to vite-plugin-ssg`);
